sql/src/main/java/org/apache/druid/sql/calcite/expression/builtin/SubstringOperatorConversion.java

```diff
@@ -45,7 +46,7 @@ public class SubstringOperatorConversion implements SqlOperatorConversion
       .operatorBuilder("SUBSTRING")
       .operandTypes(SqlTypeFamily.CHARACTER, SqlTypeFamily.INTEGER, SqlTypeFamily.INTEGER)
       .functionCategory(SqlFunctionCategory.STRING)
-      .returnTypeInference(ReturnTypes.ARG0)
+      .returnTypeInference(ReturnTypes.ARG0.andThen(SqlTypeTransforms.FORCE_NULLABLE))
```

Comment:
afaik these are appied during calcite planning ; I see some tests in `ExpressionTest` ; aren't those only excercise the native layer?